### PR TITLE
[Docs] Kops installation guide stub

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
test-docs-please

**Comment by nebril** [comment]
test-me-please

File: Documentation/install/guides/kops.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+**********************************
+Kubernetes Kops Installation Guide
+**********************************
+
+As of kops 1.9 release, Cilium is supported as CNI plugin for kops-deployed clusters.
```
Comment (ianvernon) [inline Documentation/install/guides/kops.rst]: Provide a link to kops.

Comment (ianvernon) [reply Documentation/install/guides/kops.rst]: ... Cilium is a supported CNI plugin.

File: Documentation/install/guides/kops.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Cilium needs newer kernel version than default kops images, so you need to supply an ami which will be new enough for Cilium to run on it.
```
Comment (ianvernon) [inline Documentation/install/guides/kops.rst]: ... needs a newer kernel version then the default kops images...

Which kernel version? Please specify an exact version minimum.

Comment (ianvernon) [inline Documentation/install/guides/kops.rst]: Link to AMI documentation.

File: Documentation/install/guides/kops.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+The latest stable CoreOS AMI can be found using aws cli:
+
+.. code:: bash
+
+    $ aws ec2 describe-images --region=us-east-1 --owner=595879546273 \
+        --filters "Name=virtualization-type,Values=hvm" "Name=name,Values=CoreOS-stable*" \
+        --query 'sort_by(Images,&CreationDate)[-1].{id:ImageLocation}'
+
+You also need to change default etcd version run in kops, as Cilium needs at least version 3.1.0 (kops default is still on 2.0 branch).
```
Comment (ianvernon) [inline Documentation/install/guides/kops.rst]: ... change the default etcd version used in kops...

File: Documentation/install/guides/kops.rst [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+Example Cilium-backed cluster creation command would like like as follows:
```
Comment (ianvernon) [inline Documentation/install/guides/kops.rst]: The following is an example command for creating a Cilium-backed kops cluster:

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by ianvernon** [comment]
Also, please update your commit message to be more descriptive.

File: Documentation/install/guides/kops.rst [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+Cilium needs newer kernel version than default kops images, so you need to supply an ami which will be new enough for Cilium to run on it.
+
+CoreOS images have new enough kernels for Cilium to run on them and are tested by kops developers, which makes them a perfect candidate.
+
+The latest stable CoreOS AMI can be found using aws cli:
+
+.. code:: bash
+
+    $ aws ec2 describe-images --region=us-east-1 --owner=595879546273 \
```
Comment (aanm) [inline Documentation/install/guides/kops.rst:15]: Can you provide the output the user is expected to see?

File: Documentation/install/guides/kops.rst [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Example Cilium-backed cluster creation command would like like as follows:
+
+.. code:: bash
+
+    $ export KOPS_FEATURE_FLAGS=SpecOverrideFlag #for etcd override
+    $ kops create cluster \
+    --zones <zones, e.g. "eu-central-1a"> \
+    --image 595879546273/CoreOS-stable-1576.4.0-hvm \ #check the command above to verify this is the newest CoreOS image
+    --networking cilium \
+    --override "cluster.spec.etcdClusters[*].version=3.1.11" \
```
Comment (aanm) [inline Documentation/install/guides/kops.rst:34]: out of curiosity, can we ask them to push etcd to a more recent version?

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/install/guides/kops.rst (hunk 6 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst:34]: AFAIK they plan to use etcd3 in next version. Not sure if the version will be new enough.

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-docs-please

File: Documentation/install/guides/kops.rst (hunk 5 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst:15]: Done

File: Documentation/install/guides/kops.rst (hunk 4 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: Done

File: Documentation/install/guides/kops.rst (hunk 3 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: Done

File: Documentation/install/guides/kops.rst (hunk 2 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: Done

Comment (nebril) [reply Documentation/install/guides/kops.rst]: Done

File: Documentation/install/guides/kops.rst (hunk 1 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: Done

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

File: Documentation/install/guides/kops.rst [hunk 7]
```diff
@@ -0,0 +1,35 @@
+**********************************
+Kubernetes Kops Installation Guide
+**********************************
+
+As of `kops<https://github.com/kubernetes/kops>`_ 1.9 release, Cilium is a supported CNI plugin for kops-deployed clusters.
+
+Cilium needs a newer kernel version then the default kops images provide (minimum kernel version for Cilium is 4.8), so you need to supply an `ami<https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html>`_ which will be new enough for Cilium to run on it.
```
Comment (raybejjani) [inline Documentation/install/guides/kops.rst]: in "a newer kernel version then the default kops", "then" should be "than"

File: Documentation/install/guides/kops.rst [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    $ aws ec2 describe-images --region=us-east-1 --owner=595879546273 \
+        --filters "Name=virtualization-type,Values=hvm" "Name=name,Values=CoreOS-stable*" \
+        --query 'sort_by(Images,&CreationDate)[-1].{id:ImageLocation}'
+
+    {
+        "id": "595879546273/CoreOS-stable-1576.4.0-hvm"
+    }
+
+You also need to change the default etcd version used in kops, as Cilium needs at least version 3.1.0 (kops default is still on 2.0 branch).
```
Comment (raybejjani) [inline Documentation/install/guides/kops.rst]: 3.1.0 should be 3.1, right? we use 3.1.11 below.

File: Documentation/install/guides/kops.rst (hunk 7 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: done

File: Documentation/install/guides/kops.rst (hunk 8 above)
Comment (nebril) [reply Documentation/install/guides/kops.rst]: done

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-docs-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**